### ISSUE TITLE
rpc.get_candidates return type is hideous

### ISSUE THREAD

**Issue by sharonovd** [tarantool/cartridge#226]
Currently rpc.get_candidates returns combined array and map of suitable URIs (and doc lies about it).
It renders to ugly hacks like
```
local candidates = fun.iter(rpc.get_candidates('task.roles.storage', {leader_only = true})):totable()
for _, uri in pairs(candidates) do ... end
```

Please fix public API

**Comment by rosik** [comment]
Proposal:

```lua
--- List instances suitable for performing a remote call.
--
-- @tparam string role_name
-- @tparam[opt] table opts
-- @tparam ?boolean opts.leader_only default false
-- @tparam ?boolean opts.healthy_only default true
--
-- @treturn[1] {string,...} URIs
-- @treturn[2] nil
-- @treturn[2] table Error description
cartridge.rpc_get_candidates(role_name, opts)
```

**Comment by rosik** [comment]
Resolved in #239
